listen_to_mqtt.sh

```diff
@@ -51,13 +52,17 @@ listen_to_mqtt() {
           log_notice "$(cat /share/tesla_ble_mqtt/${vin}_public.pem)"
           log_warning "Private and Public keys were generated; Next:
        1/ Remove any previously deployed BLE keys from vehicle before deploying this one
-       2/ Wake the car up with your Tesla App
-       3/ Push the button 'Deploy Key'"
+       2/ Open the Tesla App on your smartphone and make sure the car is awake.Wake the car up with your Tesla App
+       3/ In Home Assistant device Tesla_BLE_${vin}, push the button 'Deploy Key'"
           ;;
 
         deploy-key)
           log_notice "Trying to deploy the public key to vehicle..."
-          send_key $vin
+          if send_key $vin; then
+            log_info "Public key successfully sent to vin:$vin"
+          else
+            log_error "Public key did not deployed to vin:$vin"
+          fi
```

Comment:
ok, though it will log the success or errors twice. --> It could be nice to mqtt pub back to home assistant to an entitiy like "key paired" Y/N?

Reply:
I had that in mind for a split second... once mqtt or the addon restarts, the msg gets lost.

Here's another option. After send_key is successful, in order not to block the listen_mqtt loop, the script launches a background loop that runs "tesla-control ping" for ~ 5 or 10 minutes... this can tell us if the key was accepted or not. Perhaps Tesla has a timeout of 3, 5 minutes?!

Once we know the key was accepted, drop a file in same directory as the keys set ~ "{vin}.deployed".

If the end user ever delete the key,.. we could detect that and delete the dropped file so the logic (HA and docker) could just skip any commands with a message ~ "keys not deployed to the car".

Reply:
Will remove the double logging... but the main reason to add the IF is for PR 68.

```
          if send_key $vin; then
            log_info "Public key successfully sent to the car"
            ### TODO add checkVehiculeValidKey tesla-control ping or list-keys or ?!
            log_info "Setting up Home Assistant device's panel"
            setupHAAutoDiscovery $vin
          else
            log_error "Public key did not deployed to vin:$vin"
          fi
```

Reply:
Dup logging removed, Done.

Reply:
Lets leave this conversation until I add logic to call tesla-control and make sure the key was accepted and then probably drop a file instead of using mqtt for the app to know if the key is deployed or not